codeU4.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sat Jun 10 09:17:26 2017
+Exercise 4, version 1
+@author: Luiza
+"""
+
+import unittest
+import numpy as np
+
+class Tile(object):
+
+    '''Given (n by m) bool 2D array consisting of islands as True
+    and water as False
+    Counts the number of separate islands, that can be formed using 4-connectivity
+    between the land tiles
+    '''
+
+    def __init__(self, bool_matrix, rows, cols):
+
+        '''Initialization of tile as 2D numpy bool array (True/False or 0/1)
+        checks whether we have the correct size and expects boolean array elements
+
+        Args:
+            bool_matrix: boolean numpy 2D array for tile
+            rows, cols: size of the tile array
+        '''
+
+        (self.rows, self.cols) = np.shape(bool_matrix)
+        self.tile = bool_matrix
+        assert self.rows == rows or self.cols == cols, \
```

Comment:
Siim, I can not figure out the following problem: 
`assert self.rows == rows and self.cols == cols` does not work, while
```
if self.cols != cols or self.rows != rows:
            assert ValueError, "Given matrix dimensions mismatch
``` works, Do you know what might be the case?